### PR TITLE
fix: url parsing from headers if req.url is localhost

### PR DESCRIPTION
**Pull request by juliusmarminge**
this should handle almost everything i hope 😅

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* main
  * **PR #500** <a href="https://app.graphite.dev/github/pr/pingdotgg/uploadthing/500?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/pingdotgg/uploadthing/500?utm_source=stack-comment).

File: packages/uploadthing/src/internal/handler.ts [hunk 1]
```diff
@@ -336,24 +336,7 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
...
-          callbackUrl.host.includes("localhost")
-        ) {
-          console.warn(
-            [
-              "[UT] [WARN] You are using a localhost callback url in production which is not supported.",
-              "Read more and learn how to fix it here: https://uploadthing.com/faq#my-callback-runs-in-development-but-not-in-production",
-            ].join(" "),
-          );
-        }
+        const callbackUrl = resolveCallbackUrl({ config, req, url });
```
Comment (markflorkowski) [inline packages/uploadthing/src/internal/handler.ts:339]: This is only used once below, can we just inline it?

Comment (markflorkowski) [reply packages/uploadthing/src/internal/handler.ts:339]: Also is there a reason we cant just pass callbackUrl in the fetch, and need to do

`callbackUrl: callbackUrl.origin + callbackUrl.pathname,` ?

File: packages/uploadthing/src/internal/handler.ts [hunk 2]
```diff
@@ -488,6 +471,55 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
...
+  let parsedFromHeaders = (
+    getHeader(opts.req, "origin") ??
+    getHeader(opts.req, "referer") ??
+    getHeader(opts.req, "host") ??
+    getHeader(opts.req, "x-forwarded-host")
+  )?.toString();
+
+  if (parsedFromHeaders && !parsedFromHeaders.includes("http")) {
+    parsedFromHeaders =
+      (getHeader(opts.req, "x-forwarded-proto") ?? "https").toString() +
```
Comment (markflorkowski) [inline packages/uploadthing/src/internal/handler.ts:504]: L503-507 is a bit hard to parse. Mind breaking this out above, then using a template string?

**Review by markflorkowski — APPROVED** [review]

File: packages/uploadthing/src/internal/handler.ts (hunk 1 above)
Comment (juliusmarminge) [reply packages/uploadthing/src/internal/handler.ts:339]: didn't wanna pass any potential search params etc to UT

Comment (markflorkowski) [reply packages/uploadthing/src/internal/handler.ts:339]: ahh makes sense.
